c/parallel/src/for.cu

```diff
@@ -42,12 +42,25 @@ Invoke(cccl_iterator_t d_in, size_t num_items, cccl_op_t op, int /*cc*/, CUfunct
     return error;
   }
 
+  if (num_items > static_cast<size_t>(std::numeric_limits<int32_t>::max()))
+  {
+    return cudaErrorInvalidValue;
+  }
+
```

Comment:
This is surprising to me. CUB should support 64-bit sizes. Why is this limitation needed?